### PR TITLE
ci: Move CI to GitHub Actions

### PR DESCRIPTION
**Pull request by gribnoysup**
## Description

This PR moves `data-service` CI to GitHub Actions and away from Travis. This PR doesn't try to improve the config, just copies whatever Travis CI was doing

## Motivation and Context

We are consistently running out of free minutes on Travis. I'm doing this change now as I'll soon submit some changes to the lib and want the CI to work

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

### CODE REVIEW

**Review by addaleax — APPROVED** [review]

File: .github/workflows/check-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  check-and-test:
+    name: Check and Test
+    # The type of runner that the job will run on
+    runs-on: ubuntu-16.04
```
Comment (rose-m) [inline .github/workflows/check-and-test.yml:19]: nit: I saw that in the Travis CI, too, but just updated it to `20.04`... Probably feel free to just upgrade it.

File: .github/workflows/check-and-test.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      MONGODB_VERSION: ${{ matrix.mongodb_version }}
+      MONGODB_TOPOLOGY: standalone
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: Adding APT Sources
+        run: sudo -E apt-add-repository -y "ppa:ubuntu-toolchain-r/test"
```
Comment (rose-m) [inline .github/workflows/check-and-test.yml:35]: Did not know the `-E` flag 👍 
Regarding repository: When using `ubuntu-20.04` these packages installed later would already be present...

**Review by rose-m — APPROVED** [review]
More GitHub Actions 👏 
Don't know if you just want to upgrade to a newer Ubuntu version - it worked well for `compass-crud` (or maybe that's what crushed the runner :D).

File: .github/workflows/check-and-test.yml (hunk 1 above)
Comment (gribnoysup) [reply .github/workflows/check-and-test.yml:19]: Using `latest` broke CI in `connection-model` this is why I preemptively used the same version here. But the packages that were breaking the CI there are not used here, so I'll change it. I also did some digging in history and I think I'll remove it there also, seems like they were added as a workaround for some issue that is non-existent anymore

File: .github/workflows/check-and-test.yml (hunk 2 above)
Comment (gribnoysup) [reply .github/workflows/check-and-test.yml]: I don't think its really needed here, but this is what Travis does and I wanted it to be as close to it as possible (removed it all now anyway 😄)

**Pull request merged**